src/event_listener.rs

```diff
@@ -59,6 +59,21 @@ impl FlushJobInfo {
     }
 }
 
+pub struct Status {
+    result: Result<(), String>,
+    ptr: *mut DBStatusPtr,
+}
+
+impl Status {
+    pub fn reset_status(&self) {
```

Comment:
```suggestion
    pub fn reset(&self) {
```
Reset `self.result` as well.